### ISSUE TITLE
eslint server shuts down

### ISSUE THREAD

**Issue by barbalex** [Microsoft/vscode-eslint#590]
Checking the output channel reveals:

```
[Info  - 5:04:27 PM] ESLint server stopped.
[Info  - 5:04:28 PM] ESLint server running in node v8.9.3
[Info  - 5:04:28 PM] ESLint server is running.
[Info  - 5:04:29 PM] ESLint library loaded from: c:\Users\alexa\kapla3\node_modules\eslint\lib\api.js
"process.env.NODE_ENV" must be "development" to use this webpack config
[Error - 5:04:31 PM] Server process exited with code undefined. This usually indicates a misconfigured ESLint setup.
[Error - 5:04:32 PM] Connection to server got closed. Server will not be restarted.
[Info  - 5:04:32 PM] ESLint server stopped.
```

This happens every time I open the project immediately after opening the first js file.

Running `cross-env NODE_ENV=development eslint --cache --format=pretty .` on the command line works nicely.

What could I be doing wrong?

This could be related to https://github.com/Microsoft/vscode-eslint/issues/589

This is my package.json file: https://github.com/barbalex/kapla3/blob/f6cf167b503175f9ee8b31bff80329d347355d1c/package.json

More info:

* Windows 10 (up to date) and MacOs Mojave 10.14.2 (happens on both)
* project created using electron-react-boilerplate (https://github.com/electron-react-boilerplate/electron-react-boilerplate)
* vscode v1.30.0

**Comment by dbaeumer** [comment]
I am pretty sure that you might use a ESLint feature that requires a newer version of node. By default the eslint extension uses the node that ships with electron. Can you try to configure a different node version using the following setting: `eslint.runtime`. Best point it to the node version you use when running eslint on the command line.

**Comment by barbalex** [comment]
Not sure if I set this correctly. I tried:

* on Windows: `"eslint.runtime": "C:\\Program Files\\nodejs\\node.exe"`
* on MacOS: `usr/local/bin/node`

But in both cases the crash still occurs.

The mode version I use was 10.13. I just now updated to 11.4. I see that VsCode uses v8.9.3.

**Comment by barbalex** [comment]
O.k., the setting worked. I now get this error: 

```
[Info  - 5:26:37 PM] ESLint server stopped.
[Info  - 5:26:37 PM] ESLint server running in node v10.13.0
[Info  - 5:26:37 PM] ESLint server is running.
[Info  - 5:26:38 PM] ESLint library loaded from: c:\Users\alexa\kapla3\node_modules\eslint\lib\api.js
"process.env.NODE_ENV" must be "development" to use this webpack config
[Error - 5:26:40 PM] Server process exited with code undefined. This usually indicates a misconfigured ESLint setup.
[Error - 5:26:41 PM] Connection to server got closed. Server will not be restarted.
[Info  - 5:26:41 PM] ESLint server stopped.
```

**Comment by barbalex** [comment]
and after updating to node v11.4:

```
[Info  - 5:49:51 PM] ESLint server stopped.
[Info  - 5:49:52 PM] ESLint server running in node v11.4.0
[Info  - 5:49:52 PM] ESLint server is running.
[Info  - 5:49:53 PM] ESLint library loaded from: c:\Users\alexa\kapla3\node_modules\eslint\lib\api.js
"process.env.NODE_ENV" must be "development" to use this webpack config
[Error - 5:49:55 PM] Server process exited with code undefined. This usually indicates a misconfigured ESLint setup.
[Error - 5:49:56 PM] Connection to server got closed. Server will not be restarted.
[Info  - 5:49:56 PM] ESLint server stopped.
```

**Comment by barbalex** [comment]
As the error mentions misconfigured ESLint setup as probable cause: here is my config: https://github.com/barbalex/kapla3/blob/56ab81520ad8ad6310cf5864bf277eda11d5d465/.eslintrc

**Comment by dbaeumer** [comment]
@barbalex what happens if you run the eslint command directly in the terminal. Will that work?

**Comment by barbalex** [comment]
Yes:
```
PS C:\Users\alexa\kapla3> yarn lint
yarn run v1.12.3
$ cross-env NODE_ENV=development eslint --cache --format=pretty .
Done in 3.42s.
PS C:\Users\alexa\kapla3>
```

**Comment by dbaeumer** [comment]
@barbalex thanks. Can I easily reproduce this using the `https://github.com/barbalex/kapla3` repository. If so can you provide steps on how to do so?

**Comment by barbalex** [comment]
I think these should work:

1. clone `https://github.com/barbalex/kapla3` repository
2. yarn (or npm i) in it's folder
3. open a component in vscode => watch output in eslint channel
4. run `yarn lint` on the command line to check if that works (does for me)

**Comment by dbaeumer** [comment]
@barbalex this is caused by one of your scripts that are loaded into the ESLint server. It is `https://github.com/barbalex/kapla3/blob/master/internals/scripts/CheckNodeEnv.js` which calls `process.exit(2)` which then shutsdown the ESLint server. Fix should be to simply set the required env variable :-)

**Comment by barbalex** [comment]
By the way (though I am probably being unfair to Microsoft here): After 25 years of using Microsoft products this was probably the first instance of useful help I got from Microsoft.

This may sound negative but I take it as a great sign of what has been changing at Microsoft since Nadella is in charge.

**Comment by dbaeumer** [comment]
Thanks for the nice comment. I appreciate it. Have you considered using TypeScript for your Electron app. VS Code itself is written in TS and BTW half of the team sits in Zurich :-).

**Comment by barbalex** [comment]
Yeah, I have started using TypeScript in my newest project.

After first enjoying it I stopped using it because I use function components combined wich many HOC's using compose. And typing HOC's was too much of a hassle for me :-(

Little after removing TypeScript from that project react hooks came out. I refactored the project to use them. And I now wish I hadn't removed TypeScript, as typing hooks is so much easier.

I will certainly use TypeScript in the next new project.

And BTW I can see Zürich from my kitchen window...
